packages/nimble-components/build/generate-workers/source/matrix-renderer.ts

```diff
@@ -14,58 +16,69 @@ import type {
  * This setup is used in the wafer-map component to perform heavy computational duties
  */
 export class MatrixRenderer {
-    public columnIndexes = Int32Array.from([]);
-    public rowIndexes = Int32Array.from([]);
     public values = Float64Array.from([]);
     public scaledColumnIndex = Float64Array.from([]);
     public scaledRowIndex = Float64Array.from([]);
     public columnIndexPositions = Int32Array.from([]);
     public canvas!: OffscreenCanvas;
     public context!: OffscreenCanvasRenderingContext2D;
-    private scaleX: number = 1;
-    private scaleY: number = 1;
-    private baseX: number = 1;
-    private baseY: number = 1;
-    private dieDimensions: Dimensions = { width: 1, height: 1 };
-    private transform: Transform = { k: 1, x: 0, y: 0 };
-    private topLeftCanvasCorner!: { x: number; y: number };
-    private bottomRightCanvasCorner!: { x: number; y: number };
-    private readonly smallestMarginPossible: number = 20;
-    private margin: {
-        top: number;
-        right: number;
-        bottom: number;
-        left: number;
-    } = {
-            top: this.smallestMarginPossible,
-            right: this.smallestMarginPossible,
-            bottom: this.smallestMarginPossible,
-            left: this.smallestMarginPossible
-        };
-
-    public calculateXScaledIndex(columnIndex: number): number {
-        return this.scaleX * columnIndex + this.baseX + this.margin.left;
-    }
-
-    public calculateYScaledIndex(rowIndex: number): number {
-        return this.scaleY * rowIndex + this.baseY + this.margin.top;
+    private renderConfig: RenderConfig = {
+        containerDimensions: undefined,
+        dieDimensions: undefined,
+        margin: undefined,
+        verticalCoefficient: undefined,
+        horizontalCoefficient: undefined,
+        horizontalConstant: undefined,
+        verticalConstant: undefined,
+        labelsFontSize: undefined,
+        colorScale: undefined
+    };
+
+    private transformConfig: TransformConfig = {
+        transform: undefined,
+        topLeftCanvasCorner: undefined,
+        bottomRightCanvasCorner: undefined
+    };
+
+    public calculateHorizontalScaledIndex(columnIndex: number): number {
```

Comment:
Why is this and the next function `public`? The only places that are calling these methods functionally are in this class. The tests against these function seem unnecessary if we are validating the public methods that are ultimately calling these methods.